plugins/uefi-capsule/fu-uefi-backend-freebsd.c

```diff
@@ -23,67 +29,42 @@ struct _FuUefiBackend {
 
 G_DEFINE_TYPE (FuUefiBackend, fu_uefi_backend, FU_TYPE_BACKEND)
 
-static gchar *
-fu_uefi_backend_get_sysctl_string (const gchar *name)
-{
-	gsize len = 0;
-	g_autofree gchar *value = NULL;
-
-	if (sysctlbyname (name, NULL, &len, NULL, 0))
-		return NULL;
-	value = g_malloc0 (len);
-	if (sysctlbyname (name, value, &len, NULL, 0))
-		return NULL;
-	return g_steal_pointer (&value);
-}
-
-static guint64
-fu_uefi_backend_get_sysctl_uint64 (const gchar *name)
-{
-	g_autofree gchar *value = fu_uefi_backend_get_sysctl_string (name);
-	if (value == NULL)
-		return G_MAXUINT64;
-	return fu_common_strtoull (value);
-}
-
-/* yes, unsized uint_t */
-static guint
-fu_uefi_backend_get_entry_field (const gchar *sysctl_name, const gchar *field_name)
-{
-	g_autofree gchar *name = g_build_path (".", sysctl_name, field_name, NULL);
-	return (guint) fu_uefi_backend_get_sysctl_uint64 (name);
-}
-
 static FuUefiDevice *
-fu_uefi_backend_device_new (guint64 idx)
+fu_uefi_backend_device_new (struct efi_esrt_entry_v1 *entry, guint64 idx, GError **error)
 {
 	g_autoptr(FuUefiDevice) dev = NULL;
 	g_autofree gchar *fw_class = NULL;
-	g_autofree gchar *fw_class_fn = NULL;
 	g_autofree gchar *id = NULL;
-	g_autofree gchar *sysctl_name = NULL;
+	g_autofree gchar *phys_id = NULL;
+	uint32_t status;
 
-	sysctl_name = g_strdup_printf ("hw.efi.esrt.entry%u", (guint) idx);
-	fw_class_fn = g_build_path (".", sysctl_name, "fw_class", NULL);
-	fw_class = fu_uefi_backend_get_sysctl_string (fw_class_fn);
+	uuid_to_string (&entry->fw_class, &fw_class, &status);
```

Comment:
fwupd_guid_from_string?

Reply:
`entry->fw_class` isn't an array of bytes, it's of type `uuid`:
```c
struct uuid {
        uint32_t        time_low;
        uint16_t        time_mid; 
        uint16_t        time_hi_and_version;
        uint8_t         clock_seq_hi_and_reserved;
        uint8_t         clock_seq_low;
        uint8_t         node[_UUID_NODE_LEN]; 
};
```

Reply:
Related question; where is efi_esrt_entry_v1 defined?

Reply:
[Here](https://github.com/3mdeb/freebsd-src/pull/2/files#diff-572d772ebf30e43ba6125d73894f6312242f94a7d1627c06fc3ee4601947ee1eR137), it's a new structure.